src/wasm/wasm-s-parser.cpp

```diff
@@ -3027,9 +3027,14 @@ Expression* SExpressionWasmBuilder::makeArraySet(Element& s) {
 }
 
 Expression* SExpressionWasmBuilder::makeArrayLen(Element& s) {
-  // Ignore the type annotation and don't bother validating it.
-  parseHeapType(*s[1]);
-  auto ref = parseExpression(*s[2]);
+  // There may or may not be a type annotation.
+  Index childIdx = 1;
```

Comment:
We have lots of places in this file that use `i` for such a moving index. But I don't feel strongly, maybe this is clearer.